plugins/ProfessionalServices/Widgets/DismissibleWidget.php

```diff
@@ -0,0 +1,24 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\Plugins\ProfessionalServices\Widgets;
+
+use Piwik\Widget\Widget;
+
+abstract class DismissibleWidget extends Widget
+{
+    public static function getDismissibleWidgetName(): string
+    {
+        return substr(strrchr(static::class, "\\"), 1);
```

Comment:
Out of scope here but we do this in so many places in one way or another, we should consider starting something like `core/Helpers.php` and have methods like this there, e.g. `public static getClassShortName(string $class): string` etc.

Reply:
I think we should possibly rethink how we do this, I prefer having an explicit list of names and not try and calculate from the class name of PHP. It's a bit difficult to read.

However will leave it for now

Reply:
Yeah, there might be different names how to derive the name and check for the widget existence by the name. Can be done later.